### DISCUSSION TITLE
Type error: Argument of type 'StateCreator<...

### DISCUSSION THREAD

**Discussion by RowinVanAmsterdam** [pmndrs/zustand#2892] [Bug report]
### Bug Description

By following the documentation and examples, typescript will throw error below. Found similar issue [2168](https://github.com/pmndrs/zustand/discussions/2168) and fix [#2170](https://github.com/pmndrs/zustand/pull/2170) but seems like issue returned? 

 I tested it on "zustand": "^5.0.2"

```
Argument of type 'StateCreator<BearState | Partial<BearState>, [], [["zustand/devtools", never]]>' is not assignable to parameter of type 'StateCreator<BearState, [], [["zustand/devtools", never]]>'.
  Type 'StateCreator<BearState | Partial<BearState>, [], [["zustand/devtools", never]]>' is not assignable to type '(setState: { (partial: BearState | Partial<BearState> | ((state: BearState) => BearState | Partial<BearState>), replace?: false | undefined): void; (state: BearState | ((state: BearState) => BearState), replace: true): void; }, getState: () => BearState, store: StoreApi<...>) => BearState'.
    Type 'BearState | Partial<BearState>' is not assignable to type 'BearState'.
      Type 'Partial<BearState>' is not assignable to type 'BearState'.
        Types of property 'bears' are incompatible.
          Type 'number | undefined' is not assignable to type 'number'.
            Type 'undefined' is not assignable to type 'number'.
```

Code:
```
import { create } from 'zustand';
import { devtools } from 'zustand/middleware';

interface BearState {
    bears: number;
    increase: (by: number) => void;
}

const useBearStore = create<BearState>()(
    devtools(
        (set) => ({
            bears: 0,
            increase: (by) => set((state) => ({ bears: state.bears + by }))
        }),
        {
            name: 'bear-storage'
        }
    )
);
```

### Reproduction Link

Create new store with example code from docs

**Comment by dai-shi** [comment]
This works: https://tsplay.dev/NV8AZW

Can you reproduce it in the TS playground?

**Reply by RowinVanAmsterdam** [reply]
Thanks for your quick response! I looked at yours. Tried codesandbox and stackblitz, but was not able to re-produce the issue. So I restarted vs-code to make sure it wasn't the IDE. 

And this fixed the error... I feel dumb now. 


EDIT: 
Not fixed, seems like the issue returns once I rename the type. I will try to reproduce it.

**Reply by RowinVanAmsterdam** [reply]
Okay this is very weird behaviour. At least I found a way to trigger it, but still not able to reproduce it in a TS playground. 

This now seems to work:
```
import { create } from 'zustand';
import { devtools } from 'zustand/middleware';

interface BearState {
    bears: number;
    increase: (by: number) => void;
}

const useBearStore = create<BearState>()(
    devtools(
        (set) => ({
            bears: 0,
            increase: (by) => set((state) => ({ bears: state.bears + by }))
        }),
        {
            name: 'bear-storage'
        }
    )
);
```

Once I rename the interface, the error returns. Even after closing/opening again vs-code: 

```
import { create } from 'zustand';
import { devtools } from 'zustand/middleware';

interface BearStateNew {
    bears: number;
    increase: (by: number) => void;
}

const useBearStore = create<BearStateNew>()(
    devtools(
        (set) => ({
            bears: 0,
            increase: (by) => set((state) => ({ bears: state.bears + by }))
        }),
        {
            name: 'bear-storage'
        }
    )
);
```

EDIT:
- Restarting TS server > no difference
- Closing and reopening file > no difference
- Closing and reopening vs-code > no difference

**Reply by dbritto-dev** [reply]
@RowinVanAmsterdam what about type instead of interface?

**Reply by RowinVanAmsterdam** [reply]
> @RowinVanAmsterdam what about type instead of interface?

Same behaviour. Once I rename it, it throws the error again and won't go away. 

At least it seems like it's related to 'devtools'. Once I remove that, I do not get the error.

**Reply by dbritto-dev** [reply]
really weird, I wasn't able to replicate your issue

**Reply by RowinVanAmsterdam** [reply]
@dai-shi @dbritto-dev  I created a new nextjs project and have the same issue over there: [repo](https://github.com/RowinVanAmsterdam/ts-devtools-error)

**Reply by dbritto-dev** [reply]
@RowinVanAmsterdam all good I don't have any issue

<img width="1159" alt="image" src="https://github.com/user-attachments/assets/435f963d-ce6a-4523-856c-b507c1d24255" />

**Reply by RowinVanAmsterdam** [reply]
@dbritto-dev thanks for checking, you also don't see the error if you rename BearStateNew type?

**Reply by dbritto-dev** [reply]
@RowinVanAmsterdam yeah, all good 

<img width="700" alt="image" src="https://github.com/user-attachments/assets/3ce90048-f677-4e62-ae8d-1e72fd494f70" />

**Reply by dbritto-dev** [reply]
seems something else on your side

**Reply by RowinVanAmsterdam** [reply]
> seems something else on your side

I think you are right, since it's working on your side. One day later I got an update from vs-code, and now it seems to work. Even after renaming the interface/type. Thanks for helping. I will close the discussion :)
